src/Microsoft.Health.ControlPlane.CosmosDb/Features/Storage/ControlPlaneCollectionInitializer.cs

```diff
@@ -0,0 +1,26 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.ControlPlane.CosmosDb.Features.Storage.Versioning;
+using Microsoft.Health.CosmosDb.Configs;
+using Microsoft.Health.CosmosDb.Features.Storage;
+
+namespace Microsoft.Health.ControlPlane.CosmosDb.Features.Storage
+{
+    public class ControlPlaneCollectionInitializer : CollectionInitializer
+    {
+        public ControlPlaneCollectionInitializer(CosmosDataStoreConfiguration cosmosDataStoreConfiguration, ControlPlaneCollectionUpgradeManager controlPlaneCollectionUpgradeManager, ILogger<ControlPlaneCollectionInitializer> logger)
```

Comment:
Since both ControlPaneCollectionInitializer and FhirCollectionInitializer don't actually overwrite any methods and just simply passing the constructor values through, another way that you _could_ do this is do this at the IoC container. Instead of having two derived classes, you could just register CollectionInitializer like:

```
services.Add<CollectionInitializer>(sp =>
{
    var config = sp.GetService<CosmosDataStoreConfiguration>();
    var upgradeManager = sp.GetService<ControlPlaneCollectionUpgradeManager>();
    var loggerFactory = sp.GetService<ILoggerFactory>();

    return new CollectionInitializer(
        config.ControlPlaneCollectionId,
        config.RelativeDatabaseUri,
        config.RelativeControlPlaneCollectionUri,
        config.InitialControlPlaneCollectionThroughput,
        upgradeManager,
        loggerFactory.CreateLogger<CollectionInitializer>());
})
.Singleton()
.AsService<ICollectionInitializer>();
```

Since you are consuming the ICollectionInitializer as IEnumerable<ICollectionInitializer> anyways. But then again, i don't know which one is cleaner 😂

Reply:
With the separation of the collection settings I've moved to the model suggested above.